pkg/server/conn.go

```diff
@@ -1308,8 +1309,11 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	cc.lastPacket = data
 	cmd := data[0]
 	data = data[1:]
+	sqlID := cc.ctx.GetSessionVars().AllocNewSQLID()
```

Comment:
move this after line 1315?

Reply:
In my previous plan, sqlID can be an standalone ID from topSql, while internal used only now. To reduce impact when topSQL is disabled, I agree to alloc new sqlIDs when topSQL enabled.